docs/http_api.md

```diff
@@ -0,0 +1,114 @@
+# SQLD HTTP API
+
+This is the documentation for the sqld HTTP API.
+
+## Usage
+
+### The `Value` type
+
+The `Value` type represents an SQLite value. It has 4 variants:
+
+- Text: a UTF-8 encoded string
+- Integer: a 64-bit signed integer
+- Real: a 64-bits floating number
+- Blob: some binary data, encoded in base64
+- Null: the null value.
+
+All these types map to JSON straightforwardly, with two exceptions:
+- Blob: they are represented as an object with { "type": "blob", "base64": /* base64 encoded blob */}
+- Integers: they are represented as string since some languages are not able to represent them in full fidelity.
+
+### Routes
+
+#### Queries
+
+```
+POST /query
+```
+
+This endpoint supports sending batches of queries to the database. All of the statements in the batch are executed as part of a transaction. If any statement in the batch fails, an error is returned and the transaction is aborted, resulting in no change to the database.
+
+The HTTP API is stateless, which means that interactive transactions are not possible. Since all batches are executed as part of transactions, any transaction statements (e.g `BEGIN`, `END`, `ROLLBACK`...) are forbidden and will yield an error.
```

Comment:
How does this work (or not work) with savepoints? https://www.sqlite.org/lang_savepoint.html

Reply:
I think they should be supported as is, but I'll confirm when I have dig further into the issue

Reply:
It might be kind of odd for the developer to work with savepoints in a way that appears to be outside of a transaction (from their POV, since they are never allowed to be explicit about it).  I am leaning toward disallowing anything that has to do with a transaction.

The downside to this is that now they can't do anything with transactions other than implied transaction we provide.

Reply:
also, it's not like they can really perform any logic, so maybe it's not very usefull